src/atldld/utils.py

```diff
@@ -95,60 +96,44 @@ def get_image(image_id, folder=None, expression=False, downsample=0):
     Raises
     ------
     ValueError
-        If request to download the image failed.
+        If the image has a wrong format (determined by the dtype).
 
     References
     ----------
     [1] `AllenSDK API: ImageDownloadApi <https://allensdk.readthedocs.io/
     en/latest/allensdk.api.queries.image_download_api.html#allensdk.api.
     queries.image_download_api.ImageDownloadApi>`_
-
     """
     folder = folder or CACHE_FOLDER
-    folder = str(
-        folder
-    )  # this should guarantee that also LocalPath works (pytest uses it)
-
-    # Check the folder exists
-    if not os.path.exists(folder):
-        os.makedirs(folder)
-
-    # Create full path
-    additional_specifier = "_expression" if expression else ""
-    additional_specifier += f"_{downsample}"
-    path = "{}{}{}.jpg".format(folder, image_id, additional_specifier)
-
-    # Check image exists
-    if os.path.exists(path):
-        img = plt.imread(path)
-
-        if not img.dtype == np.uint8:
-            raise ValueError("The dtype needs to be uint8")
-
-        return img
-
-    else:
-        image_url = (
-            f"https://api.brain-map.org/api/v2/section_image_download/{image_id}"
-        )
-        options = []
-
+    folder = pathlib.Path(folder)
+    folder.mkdir(exist_ok=True, parents=True)
+
+    # Construct the image file name and the full path
+    file_name = f"{image_id}-{downsample}"
+    if expression:
+        file_name += "-expression"
+    image_path = folder / (file_name + ".jpg")
+
+    # Download the image if not already in the cache
+    if not image_path.exists():
+        base_url = "https://api.brain-map.org/api/v2/section_image_download"
+        url = f"{base_url}/{image_id}?downsample={downsample}"
         if expression:
-            options.append("view=expression")
-
-        if downsample:
-            options.append(f"downsample={downsample}")
+            url += "&view=expression"
 
-        image_url += "?" + "&".join(options)
+        # Download the image
+        response = requests.get(url)
+        response.raise_for_status()
```

Comment:
Cool, did not know about this one. I guess in the background it just checks the status and if it is not the right one it raises an exception, right? Or is there some more magic going on?

Reply:
Yeah, I think it's as simple as that.

In the [Quick Start docs of `requests`](https://docs.python-requests.org/en/master/user/quickstart/#response-status-codes) it says:
> If we made a bad request (a 4XX client error or 5XX server error response), we can raise it with `Response.raise_for_status()`

[The implementation of `Response.raise_for_status()`](https://docs.python-requests.org/en/master/_modules/requests/models/#Response.raise_for_status) seems pretty straightforward too.

I guess there are different ways to check if a response was OK:
- `Response.ok`
- `Response.statuse_code`
- `Response.raise_for_status()`

Here I chose not to handle any issues with requests at all and just raise. I don't know how we prefer handling this, maybe a clear common strategy would be good?

Reply:
By the way, I just checked, and `Response.ok` is implemented like that:
```python
    @property
    def ok(self):
        try:
            self.raise_for_status()
        except HTTPError:
            return False
        return True
```

Reply:
Cool! Thank you for the additional explanation!